server/src/service.rs

```diff
@@ -107,10 +107,38 @@ impl Service {
     #[zbus(out_args("secrets"))]
     pub async fn get_secrets(
         &self,
-        _items: Vec<OwnedObjectPath>,
-        _session: ObjectPath<'_>,
+        items: Vec<OwnedObjectPath>,
+        session: OwnedObjectPath,
     ) -> Result<HashMap<OwnedObjectPath, SecretInner>, ServiceError> {
-        todo!()
+        let mut secrets: HashMap<OwnedObjectPath, SecretInner> = HashMap::new();
+        let collections = self.collections.lock().await;
+
+        'outer: for collection in collections.iter() {
+            for item in &items {
+                if let Some(item) = collection.item_inner(item).await {
+                    match item.get_secret(session.clone()).await {
+                        Ok(secret) => {
+                            secrets.insert(item.path().clone(), secret.0);
+                            if secrets.len() == items.len() {
```

Comment:
What is this condition for?

Reply:
This is to avoid iterating through all the available collections (Service.Collections). If we have found the matching items to dbus call arguments, there is no need to continue iterating through remaining collections.

Reply:
Then add a small comment explaining why.

Reply:
Done.